lib/syskit/roby_app/plugin.rb

```diff
@@ -146,8 +148,64 @@ def self.setup(app)
 
                 rtt_core_model = app.default_loader.task_model_from_name("RTT::TaskContext")
                 Syskit::TaskContext.define_from_orogen(rtt_core_model, register: true)
+
+                # Log Transfer FTP Server spawned during Application#setup
+                start_tmp_root_ca
+
+                start_local_log_transfer_server(
+                    app.log_dir,
+                    @tmp_root_ca.ca_user,
+                    @tmp_root_ca.ca_password,
+                    @tmp_root_ca.signed_cert_filepath
+                )
+            end
+
+            def self.send_file_transfer_command(name, tmp_root_ca, logfile)
+                # Checks if said process server exists and is connected
+                if process_servers[name]
+                    # Establishes communication with said process server
+                    client = Syskit::RobyApp::RemoteProcesses::Client.new(
+                            host: process_server[name].host_id,
+                            port: @server_port)
+                    # Commands method log_upload_file from said process server
+                    client.log_upload_file(
+                        host: "localhost", 
+                        port: @log_transfer_port.port, 
+                        certificate: tmp_root_ca.cert_filepath, 
```

Comment:
Using a file here makes no sense. The process server is *remote*, it has no access to local files.

In other words, this should really be the certificate as a string, as you can see [here](https://github.com/rock-core/tools-syskit/blob/master/test/roby_app/remote_processes/test_remote_processes.rb#L192)

Reply:
That reminded me a question I forgot to raise. 
How could we pass the certificate created by the TmpRootCA to the remotes process servers?
It came to mind creating an empty hash in the process server code to store that certificate, passed by the TmpRootCA
when created, but it was forgotten during the Niteroi's testing week.